### PR TITLE
[Misc]Further  reduce BNB static variable

### PR DESCRIPTION
**Pull request by jeejeelee**
I have tested these changes locally using llama/mllama/minicpmv2.6/qwen.25/phi-3, all can generate reasonable results.
Thanks to @Isotr0py for his previous work, which made this PR easier.

### CODE REVIEW

File: vllm/model_executor/model_loader/loader.py [hunk 1]
```diff
@@ -989,36 +998,66 @@ def _unquantized_generator(self, hf_weights_files, use_safetensors,
...
-            if hasattr(model, 'default_bitsandbytes_target_modules'):
-                self.target_modules = model.default_bitsandbytes_target_modules
-            else:
-                self.target_modules = self.default_target_modules
-
         # Modules whose weights might have fused on disk
         # we need their output_sizes to make shard in flight correctly with TP
         self.maybe_fused_weights_modules: Dict[str, List[int]] = {}
-
+        self._get_support_modules(model)
```
Comment (mgoin) [inline vllm/model_executor/model_loader/loader.py:1060]: I think this name "support" is unclear, maybe a better one is `_get_bnb_target_modules()`?

**Review by mgoin — APPROVED** [review]
Great work I think this is a clear improvement! I just have a nit on the function name

File: vllm/model_executor/model_loader/loader.py (hunk 1 above)
Comment (jeejeelee) [reply vllm/model_executor/model_loader/loader.py]: make sense

**Comment by jeejeelee** [comment]
@mgoin It looks like  the CI failure is unrelated to this PR, maybe we need to force merge?

**Pull request merged**